src/DIRAC/Core/Utilities/ClassAd/ClassAdLight.py

```diff
@@ -3,31 +3,30 @@
 """
 
 
-class ClassAd(object):
-    def __init__(self, jdl):
+from DIRAC.Core.Utilities.ReturnValues import S_ERROR, S_OK
+
+
+class ClassAd:
+    def __init__(self, jdl: str = ""):
         """ClassAd constructor from a JDL string"""
         self.contents = {}
-        result = self.__analyse_jdl(jdl)
-        if result:
-            self.contents = result
+        if jdl:
+            result = self.__analyse_jdl(jdl)
+            if not result["OK"]:
+                raise Exception(result["Message"])
```

Comment:
Can you look for a more specialized Exception?

Reply:
We could do a SyntaxError (sketchy because if not really a Python SyntaxError...). The best way would be to implement a custom JDLSyntaxError, but given that DIRAC only uses Exceptions for the __init__ methods, i think that using Exception is fine for the moment.

Reply:
`SyntaxError` would be good enough